hil/api.py

```diff
@@ -800,7 +800,10 @@ def network_create(network, owner, access, net_id):
     else:
         if not get_network_allocator().validate_network_id(net_id):
             raise BadArgumentError("Invalid net_id")
-        allocated = False
+        if not get_network_allocator().network_id_available(net_id):
+            raise AllocationError("Network ID is in use. Please choose a "
+                                  "different net_id")
+        allocated = get_network_allocator().is_network_id_in_pool(net_id)
```

Comment:
why would it always be False? If `net_id` belongs to the vlan pool, then even though it wasn't "allocated" by HIL, we still need to mark it as `True` otherwise we can't free the network in [`network_delete`](https://github.com/CCI-MOC/hil/blob/master/hil/api.py#L838)